### PR TITLE
PWGHF: fix track selection in index skimming LfCascades

### PR DESCRIPTION
**Pull request by ZFederica**
- fix track selection in toXiPi candidateCreator (filter --> TESTBIT for CandidateType::CandV0bachelor=2)
- change filter declaration

### CODE REVIEW

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx [hunk 1]
```diff
@@ -3054,10 +3054,9 @@ struct HfTrackIndexSkimCreatorLfCascades {
   }
 
   Filter filterSelectCollisions = (aod::hf_sel_collision::whyRejectColl == 0);
-  Filter filterSelectTrackIds = (aod::hf_sel_track::isSelProng >= 4); // select tracks passing bachelor selection
```
Comment (fgrosa) [inline PWGHF/TableProducer/trackIndexSkimCreator.cxx:3057]: Here I would keep the filter, to at least remove already all the tracks that are not selected by any cut as done for 2 and 3 prongs https://github.com/AliceO2Group/O2Physics/blob/c55ebbfbe855c4774fce3e9ac5a5978ab97f4a13/PWGHF/TableProducer/trackIndexSkimCreator.cxx#L1094.

**Review by fgrosa — CHANGES_REQUESTED** [review]
Hi @ZFederica thanks a lot! I have a minor comment, then I would ask you to implement the same also for the Lc->pK0s case (`HfTrackIndexSkimCreatorCascades` struct) so that we fix it all in one go.

**Comment by ZFederica** [comment]
Hi @fgrosa, as for the struct ```HfTrackIndexSkimCreatorCascades``` I only changed the filter from isSelProng() >=4 to isSelProng > 0. The TESTBIT check was alredy in place at https://github.com/AliceO2Group/O2Physics/blob/aa3d8fe4f13bec978625c9a642e6b53c42751a28/PWGHF/TableProducer/trackIndexSkimCreator.cxx#L2764

**Review by fgrosa — APPROVED** [review]

File: PWGHF/TableProducer/trackIndexSkimCreator.cxx [hunk 2]
```diff
@@ -2667,13 +2667,13 @@ struct HfTrackIndexSkimCreatorCascades {
...
-  using SelectedCollisions = soa::Filtered<soa::Join<aod::Collisions, aod::HfSelCollision>>;
-  using FilteredTrackAssocSel = soa::Filtered<soa::Join<aod::TrackAssoc, aod::HfSelTrack>>;
-
   Filter filterSelectCollisions = (aod::hf_sel_collision::whyRejectColl == 0);
-  Filter filterSelectTrackIds = (aod::hf_sel_track::isSelProng >= 4);
+  Filter filterSelectTrackIds = (aod::hf_sel_track::isSelProng > 0);
   // Partition<MyTracks> TracksWithPVRefitAndDCA = aod::hf_sel_track::isSelProng >= 4;
 
+  using SelectedCollisions = soa::Filtered<soa::Join<aod::Collisions, aod::HfSelCollision>>;
+  using FilteredTrackAssocSel = soa::Filtered<soa::Join<aod::TrackAssoc, aod::HfSelTrack>>;
```
Comment (vkucera) [inline PWGHF/TableProducer/trackIndexSkimCreator.cxx:2675]: These type declarations should not have been moved.

**Review by vkucera — CHANGES_REQUESTED** [review]

**Pull request merged**